clang/utils/TableGen/ClangDiagnosticsEmitter.cpp

```diff
@@ -1355,7 +1357,7 @@ static void verifyDiagnosticWording(const Record &Diag) {
   if (isDigit(FullDiagText.back()) && *(FullDiagText.end() - 2) == '}') {
     // Scan backwards to find the opening curly brace.
     size_t BraceCount = 1;
-    auto Iter = FullDiagText.end() - /*}0*/ 3;
+    auto Iter = FullDiagText.end() - sizeof("}0");
```

Comment:
Does this not need a -1? Only 2 chars, but was/ends up being 3? Or is this offset because 'end' is 1 past end?

Reply:
It's because we want to get to the iterator just *before* the `}0`, so -2 gets us to `}` while -3 gets us to the character before the `}`.